File: updater.py
import os

def main():
    file_path = "files/counter.txt"
    
    try:
        with open(file_path, "r") as file:
            content = file.read().strip()
            counter_value = int(content)
    except (FileNotFoundError, ValueError):
        counter_value = 0
    
    counter_value += 1
    
    with open(file_path, "w") as file:
        file.write(str(counter_value))
    
    print("Updated File Content:")
    print(counter_value)

if __name__ == "__main__":
    main()

File: updater.go
package main

import (
	"fmt"
	"log"
	"os"
	"strconv"
)

func increase(a int) int {
	return a + 1
}

func main() {
	filePath := "files//counter.txt"

	content, err := os.ReadFile(filePath)
	if err != nil {
		log.Fatalf("Error reading file : %v", err)
	}
	counterValue, err := strconv.Atoi(string(content))
	if err != nil {
		log.Fatalf("Error converting file content to integer: %v", err)
	}

	// Modify counter
	updatedContent := strconv.Itoa(increase(counterValue))

	err = os.WriteFile(filePath, []byte(updatedContent), 0644)
	if err != nil {
		log.Fatalf("Error converting file content to integer: %v", err)
	}

	fmt.Println("Updated File Content:")
	fmt.Println(string(updatedContent))
}
